server/market/orderrouter.go

```diff
@@ -37,25 +37,25 @@ type MarketTunnel interface {
 	// MidGap returns the mid-gap market rate, which is ths rate halfway between
 	// the best buy order and the best sell order in the order book.
 	MidGap() uint64
-	// OutpointLocked should return true if the outpoint is currently a funding
-	// Coin for an active DEX order. This is required for Coin validation to
-	// prevent a user from submitting multiple orders spending the same Coin. This
+	// CoinLocked should return true if the CoinID is currently a funding Coin
+	// for an active DEX order. This is required for Coin validation to prevent
+	// a user from submitting multiple orders spending the same Coin. This
 	// method will likely need to check all orders currently in the epoch queue,
 	// the order book, and the swap monitor, since UTXOs will still be unspent
 	// according to the asset backends until the client broadcasts their
 	// initialization transaction.
 	//
-	// NOTE: This function could also potentially be handled by persistent
+	// DRAFT NOTE: This function could also potentially be handled by persistent
 	// storage, since active orders and active matches are tracked there.
-	OutpointLocked([]byte) bool
+	CoinLocked(coinID order.CoinID, assetID uint32) bool
```

Comment:
You could also pass the `DEXCoinLocker` directly to the order book router, if that makes things easier.

Reply:
OK either way.  But see what you make of my comment regarding the extra lock check in https://github.com/decred/dcrdex/pull/85#discussion_r354528345